File: lab-2/src/moves/RockTomb.java
package moves;

import ru.ifmo.se.pokemon.*;

public class RockTomb extends PhysicalMove {
    public RockTomb(){
        super(Type.ROCK, 60,95);
    }

    protected void applyOppEffects(Pokemon p){
        p.setMod(Stat.SPEED, -1);
    }

    public String describe(){
        return "is using Rock Tomb";
    }
}

File: lab-2-kotlin/src/moves/RockTomb.kt
package moves

import ru.ifmo.se.pokemon.*

class RockTomb : PhysicalMove(Type.ROCK, 60.0, 95.0) {
    override fun applyOppEffects(p: Pokemon) {
        p.setMod(Stat.SPEED, -1)
    }

    public override fun describe(): String {
        return "is using Rock Tomb"
    }
}